cmd/scheduler-estimator/app/scheduler-estimator.go

```diff
@@ -143,26 +146,62 @@ func run(ctx context.Context, opts *options.Options) error {
 	return nil
 }
 
-func serveHealthzAndMetrics(address string) {
+func serveHealthzAndMetrics(healthProbeBindAddress, metricsBindAddress string) {
+	if healthProbeBindAddress == metricsBindAddress {
+		if healthProbeBindAddress != "0" {
```

Comment:
From the implementation perspective, you have retained the capability to serve Prometheus metrics and health probes using the same TCP address. I agree, but it can also lead to inconsistent behavior compared to components that use `controller-runtime`.

Reply:
Yes, but I was thinking if we change it now, users will have to use the new flag right after the update. Otherwise, they might get `address already in use` error. If we want to make it consistent with `controller-runtime`, maybe I can add this description in the deprecated flag:

> This flag is deprecated and will be removed in release 1.12+. Use --health-probe-bind-address and metrics-bind-address instead. Note: In the next release, these two addresses cannot be the same.

and remove this functionality in the next release?

Reply:
This is a viable solution that can be implemented through a smooth upgrade process. cc @RainbowMango  @whitewindmills

Reply:
+1

Reply:
> From the implementation perspective, you have retained the capability to serve Prometheus metrics and health probes using the same TCP address. I aggre

+1

> but it can also lead to inconsistent behavior compared to components that use controller-runtime.

This may be a break change for users, and the services related to users must be adapted. I think it is unnecessary. On the back end, these components do support address reuse, and we don't necessarily have to ask the user to differentiate.

Reply:
Thanks for the feedback and insight! Could you please clarify if you are suggesting that we should remove `go serveCombined(healthProbeBindAddress)` and change the default of `metrics-bind-address` to `8080` in this PR? I want to ensure I fully understand your recommendation. Thanks!

Reply:
I am sorry, I didn't explain it clearly. What I meant was that they can use the same address to avoid a breaking change.

Reply:
@whitewindmills Do you think we should enforce that the values of `healthProbeBindAddress` and `metricsBindAddress` be different in release 1.12+?

Reply:
since we gonna make some break changes in release 1.12+, it should be consistent with other components, that is, these two addresses cannot be the same.

Reply:
only this one component can allow these two addresses to be the same, which would be weird and difficult to maintain.

Reply:
cc @RainbowMango WDYT?

Reply:
ping @RainbowMango

Reply:
Will get back on this, but now busying with KubeCon things....

Reply:
Generally I like this approach that allows the two endpoints to share the same address, for backward compatibility.
From v1.12, we can give different default values to the `--metrics-bind-address` and `--health-probe-bind-address`. And then, from v1.13 remove the combined style and force separate the two addresses just for ease of maintenance effort.

What do you think @whitewindmills @zhzhuang-zju ?
Sorry for the delay by the way.